lib/pesticide/usage.rb

```diff
@@ -4,13 +4,14 @@ class Usage
 
     def initialize(attributes = {})
       @name = attributes.delete(:name)
-      @dose = Measure.new(attributes.delete(:dose)) if attributes[:dose]
+      @dose = (attributes[:dose] && attributes[:dose] =~ /\A-?([\,\.]\d+|\d+([\,\.]\d+)?)\s*[^\s]+\z/) ? Measure.new(attributes.delete(:dose)) : attributes.delete(:dose)
       @subject = attributes.delete(:subject) || {}
       @mode = attributes.delete(:mode)
       @issue = attributes.delete(:issue)
       @pre_harvest_interval = attributes[:pre_harvest_interval].in(:day) if attributes[:pre_harvest_interval]
       @max_inputs_count = attributes[:max_inputs_count]
       if attributes[:untreated_zone_distance]
+        attributes[:untreated_zone_distance] += 'm' if attributes[:untreated_zone_distance] =~ /\A-?([\,\.]\d+|\d+([\,\.]\d+)?)\z/
```

Comment:
This should be changed to take advantage of `Measure.new`s capability to receive the unit_name as a second parameter rather than parsing the string.
You should have something like: `Measure.new(attributes[:untreated_zone_distance], :meter)` in your conditional to be more explicit and to avoid crashes in cases where `attributes[:untreated_zone_distance]` is a Numeric.